packages/renderer/src/lib/authentication/AuthActions.svelte

```diff
@@ -0,0 +1,69 @@
+<script lang="ts">
+import { faKey, faSignIn, faSignOut } from '@fortawesome/free-solid-svg-icons';
+import { DropdownMenu } from '@podman-desktop/ui-svelte';
+
+import { handleNavigation } from '/@/navigation';
+import { NavigationPage } from '/@api/navigation-page';
+
+import { authenticationProviders } from '../../stores/authenticationProviders';
+
+export let onBeforeToggle = (): void => {};
+let showMenu = false;
+
+let clientY: number;
+let clientX: number;
+export let outsideWindow: HTMLDivElement;
+
+function toggleMenu(): void {
+  onBeforeToggle();
+  showMenu = !showMenu;
+}
+
+function handleEscape({ key }: KeyboardEvent): void {
+  if (key === 'Escape') {
+    showMenu = false;
+  }
+}
+
+function onWindowClick(e: any): void {
+  showMenu = outsideWindow.contains(e.target);
+}
+
+export function onButtonClick(e: MouseEvent): void {
+  // keep track of the cursor position
+  clientY = e.clientY;
+  clientX = e.clientX;
+  toggleMenu();
+}
+</script>
+
+<svelte:window on:keyup={handleEscape} on:click={onWindowClick} />
+
+{#if showMenu}
+  <DropdownMenu.Items clientY={clientY} clientX={clientX}>
+    <DropdownMenu.Item
+      title="Manage authentication"
+      icon={faKey}
+      onClick={() => handleNavigation({ page: NavigationPage.AUTHENTICATION })} />
+
+    {#each $authenticationProviders as provider}
+      {console.error(provider)}
+      {@const sessionRequests = provider.sessionRequests ?? []}
+      {#if provider?.accounts?.length > 0}
+        {#each provider.accounts as account}
+          <DropdownMenu.Item
+            title="Sign out of {provider.displayName} ({account.label})"
+            onClick={() => window.requestAuthenticationProviderSignOut(provider.id, account.id)}
+            icon={faSignOut} />
+        {/each}
+      {/if}
+
+      {#each sessionRequests as request}
```

Comment:
I'm not comfortable with this logic, I cannot see easily the exclusion between Sign In and Sign out.
Could it be possible to write it with an if/else instead?

Reply:
I've used the same logic as in packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte file. But sure I can simplify it